### PR TITLE
Fix/5118 issue successful plan upgrade event when receive update from stigg

### PR DESCRIPTION
**Pull request by EugeneTseitlin**
Close: #5118 

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/subscription/subscription.controller.ts [hunk 1]
```diff
@@ -39,6 +45,15 @@ export class SubscriptionController {
           updateStatusDto.id,
           createSubscriptionInput
         );
+        if (createSubscriptionInput.plan === EnumSubscriptionPlan.Pro) {
+          await this.analyticsService.track({
+            userId: "unknown",
```
Comment (overbit) [inline packages/amplication-server/src/core/subscription/subscription.controller.ts:50]: can we pass the real user id?

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/core/subscription/subscription.controller.ts (hunk 1 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/subscription/subscription.controller.ts:50]: I would love to, but it doesn't arrive from webhook. I've contacted Stigg in order to find out if it's possible to pass user id.

**Review by overbit — APPROVED** [review]

File: packages/amplication-server/src/core/subscription/subscription.controller.ts [hunk 2]
```diff
@@ -39,6 +45,19 @@ export class SubscriptionController {
           updateStatusDto.id,
           createSubscriptionInput
         );
+        const userId = updateStatusDto.metadata?.userId;
+        if (
+          createSubscriptionInput.plan === EnumSubscriptionPlan.Pro &&
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/subscription/subscription.controller.ts:50]: I think this event should be triggered with every subscription creation, with the relevant metadata - for now, just "Pro" will work, but we will need to track other future plans in the same way. 
You should probably exclude "free"

File: packages/amplication-server/src/core/subscription/subscription.controller.ts [hunk 3]
```diff
@@ -39,6 +45,19 @@ export class SubscriptionController {
...
+          userId
+        ) {
+          await this.analyticsService.track({
+            userId: userId,
+            properties: {
+              workspaceId: updateStatusDto.customer.id,
+            },
+            event: EnumEventType.WorkspacePlanUpgradeCompleted,
+          });
+        }
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/subscription/subscription.controller.ts:60]: Don't we have requirements to track the other events like update, expired, downgrade, canceled?

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/core/subscription/subscription.controller.ts (hunk 3 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/subscription/subscription.controller.ts:60]: nope, but of cause now we can easily add them.

File: packages/amplication-server/src/core/subscription/subscription.controller.ts (hunk 2 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/subscription/subscription.controller.ts:50]: I thought that the point here is to register purchase that client completed by himself. Which means that "Enterprise" plan should not be registgred.

Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/subscription/subscription.controller.ts:50]: Synced with Erez, event should be issued also when workspace was upgraded to "Enterprise" plan

File: packages/amplication-server/src/core/subscription/subscription.controller.ts [hunk 4]
```diff
@@ -39,6 +45,19 @@ export class SubscriptionController {
...
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/core/subscription/subscription.controller.ts:60]: This should be in the service and not in the controller

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**